pkg/controller/helmrelease/helmreleasemgr.go

```diff
@@ -119,12 +120,12 @@ func downloadChart(client client.Client, s *appv1.HelmRelease) (string, error) {
 	return chartDir, nil
 }
 
-//GenerateManfiestString generates the manifest string for given HelmRelease
-func GenerateManfiestString(client client.Client, mgr crmanager.Manager, s *appv1.HelmRelease) (string, error) {
-	chartDir, err := downloadChart(client, s)
+//generateResourceList generates the resource list for given HelmRelease
+func generateResourceList(mgr manager.Manager, s *appv1.HelmRelease) (kube.ResourceList, error) {
+	chartDir, err := downloadChart(mgr.GetClient(), s)
```

Comment:
I think the cache issue is related to this issue,

https://github.com/kubernetes-sigs/controller-runtime/issues/180

Reply:
`mgr.Manager.GetCache().Get(ctx·context.Context,·key·client.ObjectKey,·obj·runtime.Object)` 

might resolve the cache complain.

The issue to me is, the manager build a cache for configmap, but we still query the k8s directly for the configmap.

Query directly will cause the unstructured object to be created, which somehow confused the cache.

In other words, since we waited for the cache to build up, then we can just use the cache to get object instead of query k8s directly via manager.GetClient()